app/controllers/application_controller.rb

```diff
@@ -3,30 +3,56 @@
 class ApplicationController < ActionController::Base
   include GDS::SSO::ControllerMethods
   before_action :set_request_id
-  before_action :authenticate_user!
-  before_action :set_user_instance_variable
+  before_action :authenticate
   before_action :check_service_unavailable
```

Comment:
I realise this isn't something that was changed in this PR, but I just noticed that with the order of these before actions as currently written, users will need to log in before they see the service unavailable page. Is this the behaviour we want?

Reply:
Good spot - I think `check_service_unavailable` should probably the first thing that runs, so I'll move it to the top.

Reply:
yeah I noticed this as well, I would like to do it in its own PR, one to keep the scope of this tight to basic auth and the other is because if we need to back this out we know that we're only backing out the basic auth and not undoing other good work which then gets forgotten. Thanks for the suggestion Laurence.

Reply:
Also `check_service_unavailable` could be updated to use setting config yaml file too.

Reply:
That's a really good realisation!

Reply:
That's a very good spot Al, think it's another good candidate to go in the refactor PR